### PR TITLE
Fix Mache invocations

### LINKED ISSUE #383: Explicitly specifying 'machine' in cfg file fails
I'm trying to run zppy on cori-haswell (because its life has been extended and pm is down).

Since, E3SM unified is no longer available on cori-knl login nodes (?), I'm working in an interactive session on haswell:

```
salloc --nodes 1 --qos interactive --time 04:00:00 --constraint haswell
source /global/common/software/e3sm/anaconda_envs/load_latest_e3sm_unified_cori-haswell.sh
cd ~/E3SM/QBO/scripts
```

First try
```
zppy -c post.workdir.1.cfg
Configuration file validation passed.
Traceback (most recent call last):
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.8.0_cori-haswell/bin/zppy", line 10, in <module>
    sys.exit(main())
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.8.0_cori-haswell/lib/python3.10/site-packages/zppy/__main__.py", line 84, in main
    machine_info = MachineInfo(machine=machine)
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.8.0_cori-haswell/lib/python3.10/site-packages/mache/machine_info.py", line 75, in __init__
    raise ValueError('Unable to discover machine from host name')
ValueError: Unable to discover machine from host name
```

Second try. I added the following in the [default] section of my cfg

`machine = cori-haswell`

and run zppy again

```
zppy -c post.workdir.1.cfg
Configuration file validation passed.
climo_atm_monthly_180x360_aave_0011-0020
Traceback (most recent call last):
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.8.0_cori-haswell/bin/zppy", line 10, in <module>
    sys.exit(main())
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.8.0_cori-haswell/lib/python3.10/site-packages/zppy/__main__.py", line 140, in main
    existing_bundles = ts(config, scriptDir, existing_bundles)
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.8.0_cori-haswell/lib/python3.10/site-packages/zppy/ts.py", line 61, in ts
    machine_info = MachineInfo()
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.8.0_cori-haswell/lib/python3.10/site-packages/mache/machine_info.py", line 75, in __init__
    raise ValueError('Unable to discover machine from host name')
ValueError: Unable to discover machine from host name
(e3sm_unified_1.8.0_cori-haswell) golaz@nid00286:~/E3SM/QBO/scripts> gvim /global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.8.0_cori-haswell/lib/python3.10/site-packages/zppy/ts.py
```

Note that second failure is similar to the first one, but now occurs somewhere  else:

https://github.com/E3SM-Project/zppy/blob/06c0f20dd7d5599307542f0f98dfb77b78267b64/zppy/ts.py#L61

This is clearly a bug. It should be passing the machine name that has been declared in the [default] section. It's also strange that MachineInfo() is being called here a second time.

Secondary questions for @xylar : should mache recognize cori haswell compute nodes? (Since it's going away soon, I can live without it):

```
python
>>> import socket
>>> hostname = socket.gethostname()
>>> print(hostname)
nid00286
```

### PR DESCRIPTION
**Pull request by forsyth2**
Fix Mache invocations. Resolves #383.

### CODE REVIEW

File: tests/integration/generated/test_complete_run_cori-haswell.cfg [hunk 1]
```diff
@@ -0,0 +1,162 @@
+[default]
```
Comment (forsyth2) [inline tests/integration/generated/test_complete_run_cori-haswell.cfg:1]: Remove this file before merging.

**Comment by xylar** [comment]
@forsyth2, it seems like you're creating new `MachineInfo` objects a lot.  Are these separate scripts or should a `machine_info` object just be created and passed around or shared some other way?

**Comment by forsyth2** [comment]
@xylar They're all `python` files ultimately called from `zppy/__main__.py`, so it would be possible to pass a `machine_info` instance around, yes. It would just be another parameter to clutter up the parameter lists and I didn't think creating an instance had a huge overhead cost.

**Comment by xylar** [comment]
> t would just be another parameter to clutter up the parameter lists and I didn't think creating an instance had a huge overhead cost.

It's not a big cost but it's a lot of places to modify when there's an issue, like here.  I'm guessing that's also why the machine name was supplied one time but missed all the rest.  I think it would be best from a development perspective to invoke `mache` once if you can get away with it.

File: zppy/templates/slurm_header.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #SBATCH  --qos={{ qos }}
 #SBATCH  --partition={{ partition }}
 
-{% elif machine in ['pm-cpu', 'pm-gpu'] %}
+{% elif machine in ['cori-haswell', 'pm-cpu', 'pm-gpu'] %}
```
Comment (forsyth2) [inline zppy/templates/slurm_header.sh:17]: Going to just keep this around until Cori is officially taken offline.

**Comment by forsyth2** [comment]
Ran the complete-run test on Chrysalis (since NERSC queue times are so long). Tests passed, so this should be good.

**Pull request merged**